src/HexagonAlignLoads.cpp

```diff
@@ -0,0 +1,217 @@
+#include "HexagonAlignLoads.h"
+#include "IRMutator.h"
+#include "IROperator.h"
+#include "Scope.h"
+#include "Bounds.h"
+#include "ModulusRemainder.h"
+#include "Simplify.h"
+namespace Halide {
+namespace Internal {
+using std::vector;
+
+class HexagonAlignLoads : public IRMutator {
+public:
+    HexagonAlignLoads(Target t, int vs) : target(t), vector_size(vs) {}
+private:
+    Target target;
+    enum AlignCheck { Aligned, Unaligned, NoResult };
+    // size of a vector in bytes.
+    int vector_size;
```

Comment:
This isn't really the vector size - it's the desired alignment of vector loads. On some platforms these numbers are different. Maybe rename to required_alignment